lib/Dialect/Torch/Transforms/ScalarizeShapes.cpp

```diff
@@ -746,6 +746,98 @@ class PropagateAtenItemPattern : public OpRewritePattern<AtenItemOp> {
 };
 } // namespace
 
+namespace {
+
+LogicalResult convertOpFoldResults(ImplicitLocOpBuilder &b,
+                                   SmallVector<OpFoldResult> &converted,
+                                   SmallVector<OpFoldResult> &elements,
+                                   Type inputDtype, Type resultDtype) {
+  auto inputIsInt = dyn_cast<mlir::IntegerType>(inputDtype);
+  auto resultIsInt = dyn_cast<mlir::IntegerType>(resultDtype);
+  if (!inputIsInt && !isa<mlir::FloatType>(inputDtype))
+    return failure();
+  if (!resultIsInt && !isa<mlir::FloatType>(resultDtype))
+    return failure();
+  // if dtypes are both int or both float, no conversion needed
+  if (static_cast<bool>(inputIsInt) == static_cast<bool>(resultIsInt)) {
+    converted = elements;
+    return success();
+  }
+  for (auto e : elements) {
+    auto eValue = dyn_cast<Value>(e);
+    if (eValue && resultIsInt) {
```

Comment:
<optional> Can we invert the conditionals and the loop? Thought is its a bit cleaner to see loops either converting everything to ints or everything to floats. Just a personal preference knowing that I believe it should be possible.